src/main/java/edu/cornell/kfs/fp/batch/service/impl/AdvanceDepositServiceImpl.java

```diff
@@ -0,0 +1,1202 @@
+        }
+
+        return advanceDepositDocument;
+    }
+
+    private void createNotes(AchIncomeTransaction transaction, AdvanceDepositDocument document) {
+
+        String attachmentFile = attachmentsDirectory + "/" + CuFPConstants.ADVANCE_DEPOSIT_NOTE_FILE_PREFIX
+                + document.getDocumentNumber() + "_" + new SimpleDateFormat("yyyyMMdd_HHmmss").format(dateTimeService.getCurrentDate()) + ".txt";
+        BufferedWriter os = null;
+
+        try {
+            os = new BufferedWriter(new FileWriter(attachmentFile));
+
+            for (AchIncomeNote achIncomeNote : transaction.getNotes()) {
+                os.append(achIncomeNote.getNoteText());
+                os.append("\n");
+            }
+            os.close();
+
+            Note note = documentService.createNoteFromDocument(document, "");
+            String attachmentType = null;
+            BufferedInputStream fileStream = new BufferedInputStream(new FileInputStream(attachmentFile));
+            File file = new File(attachmentFile);
+            Attachment attachment = attachmentService.createAttachment(document.getDocumentHeader(), file.getName(), "text", (int) file.length(), fileStream, attachmentType);
+            note.setAttachment(attachment);
+            attachment.setNote(note);
+            noteService.save(note);
+        } catch (Exception e) {
+            LOG.error("Error while adding notes to advance deposit documents: " + e.getMessage(), e);
+            throw new RuntimeException("Error while adding notes to the document " + document.getDocumentNumber());
+        } finally {
+            if (os != null) {
+                try {
+                    os.close();
+                } catch (IOException ie) {
+                    // not really much to do at this point
+                }
+            }
+        }
+    }
+
+    protected void createSourceAccountingLine(AchIncomeTransaction transaction, AdvanceDepositDocument advanceDepositDocument) {
+        String chart = parameterService.getParameterValueAsString(GenerateAdvanceDepositDocumentsStep.class, CuFPParameterConstants.AdvanceDepositDocument.CHART);
+        String objectCode = parameterService.getParameterValueAsString(GenerateAdvanceDepositDocumentsStep.class, CuFPParameterConstants.AdvanceDepositDocument.OBJECT_CODE);
+        String account = parameterService.getParameterValueAsString(GenerateAdvanceDepositDocumentsStep.class, CuFPParameterConstants.AdvanceDepositDocument.ACCOUNT);
+
+        IncomingWireAchMapping matchingIncomingWireAchMapping = null;
+        Collection<IncomingWireAchMapping> incomingWireAchMappings = businessObjectService.findAll(IncomingWireAchMapping.class);
+
+        for (IncomingWireAchMapping mapping : incomingWireAchMappings) {
+            List<AchIncomeNote> notes = transaction.getNotes();
+            if (doNotesMatch(mapping, notes)) {
+                matchingIncomingWireAchMapping = mapping;
+                break;
```

Comment:
I was often taught to try to avoid using "break" and "continue" within loops, but perhaps it's okay for this kind of loop. I think it's fine for now; perhaps we will learn more about this as we work through the "Clean Code" book.